name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/ComputersharePDFExtractor.java

```diff
@@ -0,0 +1,172 @@
+                                idxSingleDate = idxSingleDate == -1 && singleDatePattern.matcher(lines[i]).matches() ? i
+                                                : idxSingleDate;
+                                idxDoubleDate = idxDoubleDate == -1 && doubleDatePattern.matcher(lines[i]).matches() ? i
+                                                : idxDoubleDate;
+                            }
+                            if (idxSingleDate > 3 && idxDoubleDate > 4 && (idxDoubleDate - 2) == idxSingleDate)
+                            {
+
+                                final Matcher mCusip = cusipPattern.matcher(lines[idxSingleDate - 2]);
+                                final Matcher mTickerSymbol = tickerSymbolPattern.matcher(lines[idxSingleDate - 3]);
+                                if (mCusip.matches() && mTickerSymbol.matches())
+                                {
+                                    context.put(PATTERN_KEY_TICKERSYMBOL, mTickerSymbol.group(PATTERN_KEY_TICKERSYMBOL));
+                                    context.put(PATTERN_KEY_WKN, mCusip.group(PATTERN_KEY_WKN));
+                                }
+                                else
+                                {
+                                    throw new IllegalArgumentException("Cannot retrieve security information");
+                                }
+                            
+                            }
+                        });
+
+        this.addDocumentTyp(type);
+        
+
+        // define first block to process
+        final Block purchaseBlocks = new Block(BLOCK_IDENTIFIER_PURCHASE);
+        type.addBlock(purchaseBlocks);
+        
+        // create source
+        final Transaction<BuySellEntry> purchaseTransaction = new Transaction<>();
+        purchaseBlocks.set(purchaseTransaction);
+        purchaseTransaction.subject(() -> {
+            BuySellEntry entry = new BuySellEntry();
+            entry.setType(PortfolioTransaction.Type.BUY);
+            return entry;
+        });
+        purchaseTransaction.oneOf(section -> section
+                        .attributes(PATTERN_KEY_DATE, PATTERN_KEY_AMOUNT, PATTERN_KEY_SHARES, PATTERN_KEY_FEE)
+                                        .documentContext(PATTERN_KEY_TICKERSYMBOL, PATTERN_KEY_WKN)
+                        .match(PATTERN_PURCHASE_ENTRY)//
+                                        .assign((t, v) -> {
+                            v.put(PATTERN_KEY_CURRENCY, "USD");
+                            t.setDate(asDate(v.get(PATTERN_KEY_DATE)));
+                                            t.setCurrencyCode(asCurrencyCode("USD"));
+                            t.setAmount(asAmount(v.get(PATTERN_KEY_AMOUNT)));
+                            t.setShares(asShares(v.get(PATTERN_KEY_SHARES)));
+                                            final Security security = getOrCreateSecurity(v);
+                                            security.setCurrencyCode(asCurrencyCode("USD"));
```

Comment:
This is not allowed.
The security matched by the importer might already exist in the data file of the user.
Therefore we must not change the currency (also considering that the security already has transactions and historical prices attached that assume another currency).

I will merge the importer but will change this before.